tests/subsys/emds/emds_flash/src/main.c

```diff
@@ -1,15 +1,16 @@
 /*
- * Copyright (c) 2022 Nordic Semiconductor ASA
+ * Copyright (c) 2025 Nordic Semiconductor ASA
```

Comment:
General comment. May be seperate PR. 

Since how `emds_flash_allocate_snapshot` only fills some fields of the given `allocated_snapshot` pointer, but not all (like fresh_cnt), the test also relies on this and manages `fresh_cnt` manually. I feel, we should document this in doc string for `emds_flash_allocate_snapshot`, may be in a seperate PR. So that later it is easy for maintainance. Something like this:
"This API will only populate ...blah fields. Blah field must be handled by the caller before writing the snapshot to non-volatile memory".